### PR TITLE
Fix #23165: Update StreamConverters.fromInputStream docs to reflect reality

### LINKED ISSUE #23165: Inaccurate documentation on StreamConverters.fromInputStream
`StreamConverters.fromInputStream`'s [implementation](https://github.com/akka/akka/blob/master/akka-stream/src/main/scala/akka/stream/impl/io/InputStreamPublisher.scala#L56) emits _up to_ `chunkSize`-length`ByteStream` elements, depending on the availability of the input stream each time it blocks. However, the documentation says:

_Emitted elements are chunkSize sized [[akka.util.ByteString]] elements, except the final element, which will be up to chunkSize in size._

The documentation should be updated to reflect reality.

### PR DESCRIPTION
**Pull request by jypma**
Fixes #23165 by mentioning that each element emitted by `fromInputStream` will be
UP TO the configured chunk size, rather than EXACTLY the emitted chunk size
(which was implied by the previous documentation).

### CODE REVIEW

**Review by 2m — APPROVED** [review]

**Comment by patriknw** [comment]
OK TO TEST

File: akka-stream/src/main/scala/akka/stream/javadsl/StreamConverters.scala [hunk 1]
```diff
@@ -89,8 +89,9 @@ object StreamConverters {
 
   /**
    * Creates a Source from an [[java.io.InputStream]] created by the given function.
-   * Emitted elements are `chunkSize` sized [[akka.util.ByteString]] elements,
-   * except the final element, which will be up to `chunkSize` in size.
+   * Emitted elements are up to `chunkSize` sized [[akka.util.ByteString]] elements.
+   * The actual size of the emitted elements depends on how much the incoming stream
+   * delivers on each invocation of [[read()]].
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/StreamConverters.scala]: pretty sure this syntax results in a warning and broken link. `[[]]` is used to generate links, and here scaladoc has no idea what "read" is.

Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/StreamConverters.scala]: How about we phrase like 

> The actual size of emitted elements depends how much data the underlying [[java.io. InputStream]] returns on each read invocation. Such chunk will never be larger than `chunkSize` though.

**Review by ktoso — COMMENTED** [review]
I would rephrase it a bit, and avoid the scaladoc warning

File: akka-stream/src/main/scala/akka/stream/javadsl/StreamConverters.scala (hunk 1 above)
Comment (jypma) [inline akka-stream/src/main/scala/akka/stream/javadsl/StreamConverters.scala]: I like it, I'll update.

**Comment by jypma** [comment]
Rephrased as per @ktoso 's suggestion.

**Comment by jypma** [comment]
Rephrased as per @ktoso 's suggestion.

File: akka-stream/src/main/scala/akka/stream/javadsl/StreamConverters.scala [hunk 2]
```diff
@@ -104,8 +106,10 @@ object StreamConverters {
 
   /**
    * Creates a Source from an [[java.io.InputStream]] created by the given function.
-   * Emitted elements are [[ByteString]] elements, chunked by default by 8192 bytes,
-   * except the last element, which will be up to 8192 in size.
+   * Emitted elements are up to 8192 bytes sized [[akka.util.ByteString]] elements.
+   * The actual size of emitted elements depends how much data the underlying
+   * [[java.io. InputStream]] returns on each read invocation. Such chunks will
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/StreamConverters.scala]: whoops! please fix the space after `io.` :)

**Review by ktoso — COMMENTED** [review]
All good after stray space in code link fixed :)

**Review by ktoso — APPROVED** [review]
LGTM, anyone else want to sanity check? I fixed the spaces so please squash when mergign

**Comment by jypma** [comment]
Thanks @ktoso :-)

**Review by johanandren — APPROVED** [review]
